packages/cli/src/utils/compression.util.ts

```diff
@@ -0,0 +1,198 @@
+import * as fflate from 'fflate';
+import { promisify } from 'util';
+import { readFile, readdir, writeFile, mkdir, stat } from 'fs/promises';
+import * as path from 'path';
+
+const unzip = promisify(fflate.unzip);
+const zip = promisify(fflate.zip);
+
+// Reuse the same compression levels as the Compression node
+const ALREADY_COMPRESSED = [
+	'7z',
+	'aifc',
+	'bz2',
+	'doc',
+	'docx',
+	'gif',
+	'gz',
+	'heic',
+	'heif',
+	'jpg',
+	'jpeg',
+	'mov',
+	'mp3',
+	'mp4',
+	'pdf',
+	'png',
+	'ppt',
+	'pptx',
+	'rar',
+	'webm',
+	'webp',
+	'xls',
+	'xlsx',
+	'zip',
+];
+
+export interface CompressionOptions {
+	level?: fflate.ZipOptions['level'];
+	exclude?: string[];
+	includeHidden?: boolean;
+}
+
+export interface DecompressionOptions {
+	overwrite?: boolean;
+	exclude?: string[];
+}
+
+/**
+ * Sanitize file path to prevent zip slip attacks
+ * Ensures the resolved path stays within the output directory
+ */
+function sanitizePath(fileName: string, outputDir: string): string {
+	// Normalize the path and resolve any relative path components
+	const normalizedPath = path.normalize(fileName);
+
+	// Join with output directory and resolve to get absolute path
+	const resolvedPath = path.resolve(outputDir, normalizedPath);
+	const resolvedOutputDir = path.resolve(outputDir);
+
+	// Check if the resolved path is within the output directory
+	if (
+		!resolvedPath.startsWith(resolvedOutputDir + path.sep) &&
+		resolvedPath !== resolvedOutputDir
+	) {
+		throw new Error(
+			`Path traversal detected: ${fileName} would be extracted outside the output directory`,
+		);
+	}
+
+	return resolvedPath;
+}
+
+/**
+ * Compress a folder into a ZIP archive
+ * Reuses the same patterns as the Compression node
+ */
+export async function compressFolder(
+	sourceDir: string,
+	outputPath: string,
+	options: CompressionOptions = {},
+): Promise<void> {
+	const { level = 6, exclude = [], includeHidden = false } = options;
+
+	const zipData: fflate.Zippable = {};
+
+	await addDirectoryToZip(sourceDir, '', zipData, { exclude, includeHidden, level });
+
+	// Ensure output directory exists
+	const outputDir = path.dirname(outputPath);
+	await mkdir(outputDir, { recursive: true });
+	const buffer = await zip(zipData);
```

Comment:
This seems to load in the whole exported entities into memory, for large databases this might become a problem. An improvement could be streaming the entries into the zip file. We could do this as a follow up.